docs/getting_started/basic/02_arguments_and_results.ipynb

```diff
@@ -5,51 +5,208 @@
+    "            url=os.environ.get(\"QISKIT_IBM_URL\"),\n",
+    "        )\n",
+    "        backend = service.backend(backend_name)\n",
+    "        print(\n",
+    "            f\"[main] Runtime service initialized from env and backend \"\n",
+    "            f\"resolved (name={backend.name})\"\n",
+    "        )\n",
+    "    except QiskitBackendNotFoundError as e:\n",
+    "        raise ValueError(f\"The backend named {backend_name} couldn't be found.\") from e\n",
+    "    except Exception as e:\n",
+    "        raise ValueError(\n",
+    "            f\"`QiskitRuntimeService` couldn't be initialized with os environment variables: {e}.\"\n",
+    "        ) from e\n",
     "\n",
-    "sampler = Sampler()\n",
+    "# ----- transpile -----\n",
+    "# Match the run to the backend and transpile\n",
+    "print(\"[main] Building preset pass manager (optimization_level=3)...\")\n",
+    "pm = generate_preset_pass_manager(backend=backend, optimization_level=3)\n",
+    "print(\"[main] Transpiling circuit...\")\n",
+    "isa_circuit = pm.run(circuit)\n",
+    "print(\"[main] Transpilation complete\")\n",
     "\n",
-    "quasi_dists = sampler.run(circuit).result().quasi_dists\n",
+    "# ----- execute -----\n",
+    "# Execute and collect counts\n",
+    "print(f\"[main] Executing circuit on backend (name={backend.name})...\")\n",
+    "sampler = Sampler(backend)\n",
+    "job_result = sampler.run([isa_circuit]).result()\n",
+    "print(\"[main] Sampler execution complete\")\n",
     "\n",
-    "print(f\"Quasi distribution: {quasi_dists[0]}\")\n",
+    "# Extract counts\n",
+    "quasi_dists = job_result[0].data.meas.get_counts()\n",
+    "print(\"[main] Extracted counts from result\")\n",
     "\n",
+    "# ----- persist -------\n",
     "# saving results of the execution\n",
-    "save_result({\n",
-    "    \"quasi_dists\": quasi_dists[0]\n",
-    "})\n",
+    "save_result({\"quasi_dists\": quasi_dists})\n",
+    "print(\"[main] Results saved.\")\n",
+    "\n",
     "```\n",
     "\n",
-    "As you can see, the circuit construction is not inside the function anymore. Instead, we parse the arguments by calling the [get_arguments](https://qiskit.github.io/qiskit-serverless/stubs/qiskit_serverless.serializers.get_arguments.html#qiskit_serverless.serializers.get_arguments) function."
+    "This function resolves the backend through a `QiskitRuntimeService`. For real IBM Quantum backends, the service can be provided explicitly as an argument (recommended for notebooks and local runs) or constructed dynamically from environment variables. These variables—such as `QISKIT_IBM_CHANNEL`, `QISKIT_IBM_TOKEN`, `QISKIT_IBM_INSTANCE`, and `QISKIT_IBM_URL` —are typically set by your DevOps or cloud team during deployment (for example, in Kubernetes manifests or Docker Compose). When present, the function uses them to authenticate and configure the runtime service without hardcoding credentials in the code.\n",
```

Comment:
Thanks for this :) I still refer to my question above regarding when can the service be passed as an argument, this comment makes me think I missed something.